File: leetcode/0150. Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        ops = {"+": lambda x, y : x + y,
               "-": lambda x, y : y - x,
               "*": lambda x, y : x * y,
               "/": lambda x, y : int(y / x)}
        
        stack = list()

        for token in tokens:
            if token in ops:
                stack.append(ops[token](stack.pop(), stack.pop()))
            else:
                stack.append(int(token))

        return stack[0]

File: leetcode/0150. Evaluate Reverse Polish Notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack();
        for (String token : tokens) {
            if(token.length() == 1 && !(token.charAt(0) >= '0' && token.charAt(0) <= '9')){
                int b = stack.pop();
                int a = stack.pop();

                switch(token.charAt(0)){
                    case '+':
                        stack.push(a + b);
                        break;
                    case '-':
                        stack.push(a - b);
                        break;
                    case '*':
                        stack.push(a * b);
                        break;
                    case '/':
                        stack.push(a / b);
                        break;
                }
            } else {
                stack.push(Integer.parseInt(token));
            }
        }
        return stack.peek();
    }
}